### PR TITLE
Make virtctl helpers unexported and move them to testing package

### LINKED ISSUE #12891: some virtctl unit tests are coupled to local kubeconfig
**What happened**:
having a ~/.kube/config messes up virtctl unit tests
since the namespace gets picked up from there.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
for example, login into a remote cluster using local kubectl and then run vmexport unit tests using
`go test`

**Additional context**:
```
• [FAILED] [0.001 seconds]
vmexport VMExport succeeds [It] Succesfully create VirtualMachineExport with TTL
/home.../kubevirt/pkg/virtctl/vmexport/vmexport_test.go:540

  [FAILED] Unexpected call to *kubecli.MockKubevirtClient.VirtualMachineExport([akalenyu]) at /.../kubevirt/pkg/virtctl/vmexport/vmexport.go:428 because: 
  expected call at /.../kubevirt/vendor/kubevirt.io/client-go/kubecli/generated_mock_kubevirt.go:217 doesn't match the argument at index 0.
  Got: akalenyu (string)
  Want: is equal to default (string)
```

**Environment**:
- KubeVirt version (use `virtctl version`): N/A
- Kubernetes version (use `kubectl version`): N/A
- VM or VMI specifications: N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does

This introduces the pkg/virtctl/testing package that provides helpers to
run virtctl in unit tests. The unit tests of virtctl are updated to use
this package instead of tests/clientcmd to avoid importing test code in
production code. Places that used clientcmd.NewVirtctlCommand before were
updated to use testing.NewRepeatableVirtctlCommand instead.

Move the helpers to run virtctl in functional tests from
tests/clientcmd/virtctl.go to tests/virtctl/cmd.go and update the
functional tests of virtctl to make use of the unexported helpers. This
prevents importing these helpers in other places where they should not
be used.

This also addresses a small bug in newVirtctlCommand, that was checking
the Value interface instead of the flag returned by flag.Lookup for nil.

Ultimately, this achieves a clear separation of production and test code in virtctl by avoiding imports of test code into the virtctl production code.

Before this PR:

virtctl production code imports test code.

After this PR:

virtctl production code and test code are clearly separated from each other.

Fixes #12891

The helpers in unit tests no longer pick up the `kubeconfig` and `server` flags. This should fix the issue mentioned in #12891.

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Special notes for your reviewer

Merge after #13378 and #13384

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/virtctl/testing/testing.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package testing
```
Comment (fossedihelm) [inline pkg/virtctl/testing/testing.go:1]: missing header

**Review by fossedihelm — COMMENTED** [review]
Thank you!
One leftover :)

File: pkg/virtctl/testing/testing.go (hunk 1 above)
Comment (0xFelix) [reply pkg/virtctl/testing/testing.go:20]: Addressed!

**Review by fossedihelm — APPROVED** [review]
Thank you!

**Review by xpivarc — COMMENTED** [review]
/approve

**Review by fossedihelm — APPROVED** [review]

**Pull request merged**